### PR TITLE
[CELEBORN-1545] Add Tez plugin skeleton and dag app master

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
Add tez plugin skeleton

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
No tests needed now.

### CODE REVIEW

File: client-tez/tez-shaded/pom.xml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  <parent>
+    <groupId>org.apache.celeborn</groupId>
+    <artifactId>celeborn-parent_${scala.binary.version}</artifactId>
+    <version>${project.version}</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>celeborn-client-tez-shaded_${scala.binary.version}</artifactId>
+  <packaging>jar</packaging>
+  <name>Celeborn Shaded Client for MapReduce</name>
```
Comment (SteNicholas) [inline client-tez/tez-shaded/pom.xml:29]:
```suggestion
  <name>Celeborn Shaded Client for Tez</name>
```

File: client-tez/tez/pom.xml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  <parent>
+    <groupId>org.apache.celeborn</groupId>
+    <artifactId>celeborn-parent_${scala.binary.version}</artifactId>
+    <version>${project.version}</version>
+    <relativePath>../../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>celeborn-client-tez_${scala.binary.version}</artifactId>
+  <packaging>jar</packaging>
+  <name>Celeborn Client for MapReduce</name>
```
Comment (SteNicholas) [inline client-tez/tez/pom.xml:29]:
```suggestion
  <name>Celeborn Client for Tez</name>
```

**Comment by SteNicholas** [comment]
@FMX, the CI workflow should add dependency check for client-tez module.

File: client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+          .addShutdownHook(new DAGAppMasterShutdownHook(appMaster), SHUTDOWN_HOOK_PRIORITY);
+
+      // log the system properties
+      if (Logger.isInfoEnabled()) {
+        String systemPropsToLog = TezCommonUtils.getSystemPropertiesToLog(conf);
+        if (systemPropsToLog != null) {
+          Logger.info(systemPropsToLog);
+        }
+      }
+
```
Comment (SteNicholas) [inline client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java:350]: There are some succeed tasks which may be recomputed in local shuffle, for example, node manager is unhealthy, reducer could not fetch the shuffle data from mapper which has already run successfully in unhealthy node manager. On the contrary, the shuffle data is stored in celeborn, which causes that recomputing the succeeded task which have run in unhealthy node is unnecessary. Meanwhile, `TezConfiguration.TEZ_AM_NODE_UNHEALTHY_RESCHEDULE_TASKS` could set to false for this situation.

Comment (FMX) [reply client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java:350]: I have done this.  Check the code https://github.com/apache/celeborn/blob/3c15db953d57a017b4e4fdc4c3cfe1956c48ae42/client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java#L292-L294

**Review by SteNicholas — APPROVED** [review]
LGTM. Please add dependency check for client-tez module in CI.

File: pom.xml [hunk 4]
```diff
@@ -1744,6 +1744,160 @@
...
 
+    <profile>
+      <id>tez</id>
+      <modules>
+        <module>client-tez/tez</module>
+        <module>client-tez/tez-shaded</module>
+        <module>tests/mr-it</module>
+      </modules>
+      <properties>
+        <tez.version>0.9.1</tez.version>
```
Comment (RexXiong) [inline pom.xml:1755]: use latest tez 0.9.x version

File: client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+      Options opts = new Options();
+      opts.addOption(
+          TezConstants.TEZ_SESSION_MODE_CLI_OPTION,
+          false,
+          "Run Tez Application Master in Session mode");
+
+      CommandLine cliParser = new GnuParser().parse(opts, args);
+      boolean sessionModeCliOption = cliParser.hasOption(TezConstants.TEZ_SESSION_MODE_CLI_OPTION);
+
+      Logger.info(
```
Comment (RexXiong) [inline client-tez/tez/src/main/java/org/apache/tez/dag/app/CelebornDagAppMaster.java:267]: use {} placeholder for logging

**Review by RexXiong — COMMENTED** [review]
make-distribution.sh/style.yml, etc.  also need change to support build/check tez

**Comment by RexXiong** [comment]
Merge to branch:  branch-tez-support

**Pull request merged**